packages/zwave-js/src/lib/commandclass/BarrierOperatorCC.ts

```diff
@@ -0,0 +1,157 @@
+import {
+	CCCommand,
+	CCCommandOptions,
+	CommandClass,
+	commandClass,
+	expectedCCResponse,
+	implementedVersion,
+	CommandClassDeserializationOptions,
+	gotDeserializationOptions,
+	ccValue,
+} from "./CommandClass";
+import { CommandClasses, parseBitMask, validatePayload } from "@zwave-js/core";
+import type { Driver } from "../driver/Driver";
+import { expectedResponse } from "../message/Message";
+
+// All the supported commands
+export enum BarrierOperatorCommand {
+	Set = 0x01,
+	Get = 0x02,
+	Report = 0x03,
+	CapabilitiesGet = 0x04,
+	CapabilitiesReport = 0x05,
+	EventSignalSet = 0x06,
+	EventSignalGet = 0x07,
+	EventSignalReport = 0x08,
+}
+
+
+// All possible positions for Report
+export enum BarrierReportState {
+	Closed = 0x00,
+	Closing = 0xFC,
+	Stopped = 0xFD,
+	Openning = 0xFE,
+	Open = 0xFF
+}
+
+/**
+ * @publicAPI
+ */
+export enum BarrierPosition {
+	Closed = 0x00,
+	Closing = 0xFC,
+	Stopped = 0xFD
+	Openning = 0xFE,
+	Open = 0xFF,
+}
+
+// @publicAPI
+export enum SignalType {
+	"Not Supported" = 0x00,
+	Audible = 0x01,
+	Visual = 0x02,
+}
+
+@commandClass(CommandClasses.BarrierOperator)
+@implementedVersion(1)
+export class BarrierOperatorCC extends CommandClass {
+	declare ccCommand: BarrierOperatorCommand;
+}
+
+// To be reviewed as CC should only accept 0x00 or 0xFF, if more work needed
+@CCCommand(BarrierOperatorCommand.Set)
+export class BarrierOperatorCCSet extends BarrierOperatorCC {
+	public constructor(
+		driver: Driver,
+		options: CommandClassDeserializationOptions | BarrierOperatorCCSetOptions,
```

Comment:
BarrierOperatorCCSetOptions is not defined. If you use the `zwcccmd` snippet, there should be an interface definition above the class.

Reply:
I already have found this and some more, defined some missing ones.